lib/engine-addon.js

```diff
@@ -362,6 +362,18 @@ module.exports = {
       }).filter(Boolean);
     };
 
+    /**
+     * Determines if ember-engines should use the compact-reexports Babel plugin.
+     */
+    options.shouldCompactReExports = function() {
+      var VersionChecker = require('ember-cli-version-checker');
+      var checker = new VersionChecker(this);
+
+      return checker.for('ember-cli', 'npm').satisfies('>=2.13.0') &&
+        checker.for('ember-cli-babel', 'npm').satisfies('>=6.0.0') &&
```

Comment:
This isn't quite right (sadly).  This will check if the top level project is using ember-cli-babel@6, but not if _this_ engine is (in its own `package.json`).  I'm working on refactoring ember-cli-version-checker to inherently support this as a concept (I need it for exactly this reason also), but it does not yet.